### ISSUE TITLE
feature request: Supply partitionKey for KinesisExport

### ISSUE THREAD

**Issue by obiwabrakenobi** [aws-greengrass/aws-greengrass-stream-manager-sdk-python#15]
Hi,

is it possible to submit a partition key for putting messages to kinesis?

**Comment by MikeDombo** [comment]
No it is not possible to specify it. Stream manager will always choose a random partition key to distribute the load.

**Comment by obiwabrakenobi** [comment]
Yes I saw that. We are heavily using stream manager and would benefit submitting a partition key as we need to sort data afterwards because it is distributed across multiple shards. Are there any plans to include a partition key?

**Comment by MikeDombo** [comment]
At this time there is no plan to include that.

Can you sort based on the value of the payload rather than the partition? You can include any data you want within the payload which is uploaded to Kinesis.

**Comment by obiwabrakenobi** [comment]
Yes we have a Timestamp. We are processing the data with flink from 6 shards and we need to buffer the incoming events for 1 minute until we have read every value. With a partition key we could remove this kind of buffering. I understand the even distribution but in our case that introduces a unwanted delay

**Comment by MikeDombo** [comment]
Closing this issue, we currently do not support partition key, but it is something that we will consider for future improvements.

Thank you.

**Comment by jbutler** [comment]
Re-opening. This is a feature request
